mapie/classification.py

```diff
@@ -341,33 +534,124 @@ def predict(
-                    self.scores_,
-                    ((n + 1) * (_alpha)) / n,
-                    interpolation="lower"
+                    self.conformity_scores_,
+                    ((n + 1) * (1 - _alpha)) / n,
+                    interpolation="higher"
                 ) for _alpha in alpha_
             ])
-            prediction_sets = np.stack([
-                y_pred_proba > (quantile)
-                for quantile in self.quantiles_
-            ], axis=2)
+            if self.method == "score":
+                prediction_sets = np.stack(
+                    [
+                        y_pred_proba > 1 - quantile
+                        for quantile in self.quantiles_
+                    ],
+                    axis=2,
+                )
+            elif self.method == "cumulated_score":
+                # sort labels by decreasing probability
+                index_sorted = np.fliplr(np.argsort(y_pred_proba, axis=1))
+                # sort probabilities by decreasing order
+                y_pred_proba_sorted = np.take_along_axis(
+                    y_pred_proba, index_sorted, axis=1
+                )
+                # get sorted cumulated score starting from 0
+                y_pred_proba_sorted_cumsum = np.cumsum(
+                    y_pred_proba_sorted, axis=1
+                )
+                # get the index of the last included label
+                if (
+                    (include_last_label is True) or
+                    (include_last_label == 'randomized')
+                ):
+                    y_pred_proba_sorted_last = np.stack([
+                        np.argmin(
+                            np.ma.masked_less_equal(
+                                y_pred_proba_sorted_cumsum,
+                                quantile
+                            ),
+                            axis=1
+                        )
+                        for quantile in self.quantiles_
+                    ], axis=1)
+                elif (include_last_label is False):
+                    y_pred_proba_sorted_last = np.stack([
+                        np.argmax(
+                            np.ma.masked_greater(
+                                y_pred_proba_sorted_cumsum,
+                                quantile
+                            ),
+                            axis=1
+                        )
+                        for quantile in self.quantiles_
+                    ], axis=1)
+                else:
+                    raise ValueError(
+                        "Invalid include_last_label argument. "
+                        "Should be a boolean or 'randomized'."
+                    )
+                y_proba_last = np.stack(
+                    [
+                        y_pred_proba_sorted[
+                            np.arange(len(y_pred_proba_sorted)),
+                            y_pred_proba_sorted_last[:, iq]
+                        ] for iq, _ in enumerate(self.quantiles_)
```

Comment:
Sauter une ligne juste avant le `for` par souci d'homogénéité avec les autres lignes de code

Reply:
fait